frontend/src/pages/facemash.js

```diff
@@ -30,75 +29,78 @@ class FaceMash extends Component<Props> {
   constructor(props) {
     super(props)
     this.state = {
-      results: this.props.result,
       error: false,
       message: ''
     }
   }
 
-  static async getInitialProps({ query }) {
-    // check whether query.id is real candidate
-    // TODO: do not fetch again, if user hasn't responded to previous match
-    try {
-      const { result } = await getCandidateMatch()
-      if (result === undefined) {
-        return { error: 'Bad Request' }
-      }
-      return { result }
-    } catch (err) {
-      console.log('Candidate Page error: ', err.message)
-      return { error: 'Bad Request' }
+  async getNewMatch() {
+    const { result } = await getCandidateMatch()
+    if (result === undefined) {
+      console.log('Could not get new FaceMash match')
```

Comment:
if `result` is undefined, the function still continues after logging that, meaning it will fail in line 44. maybe alert the user that it wasn't able to get a match by rendering a placeholder saying `Error: Couldn't get new FaceMash match. Please refresh` or using a popup/alert/toast?